### PR TITLE
web service functioning

### PR DESCRIPTION
**Pull request by AdiTelem**
Transferred Web DB logic to a bound service,
communication between notes data structure and db moved to a repository.

Known Bugs:
List might not load until a not is edited/deleted/created - to be fixed

### CODE REVIEW

File: app/src/main/java/com/example/notes/MainActivity.kt [hunk 1]
```diff
@@ -1,30 +1,68 @@
...
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+
         enableEdgeToEdge()
         setContent {
             NotesTheme {
-                NotesAppNavigation()
+                var navigationViewModel: NavigationViewModel = viewModel(factory = NavigationViewModel.Factory)
```
Comment (jazzyjester) [inline app/src/main/java/com/example/notes/MainActivity.kt]: @AdiTelem `var` is needed when changing the variable. can be `val`

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -5,18 +5,23 @@
...
         android:label="@string/app_name"
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/Theme.Notes"
-        tools:targetApi="31"
-        tools:ignore="UnusedAttribute">
+        tools:ignore="UnusedAttribute"
+        tools:targetApi="31">
+
+        <service android:name=".WebService"/>
```
Comment (jazzyjester) [inline app/src/main/AndroidManifest.xml:23]: @AdiTelem Class name WebService is very general.
At least the name should suggests what is the purpose of the service.
like "DeleteNoteService" etc...

File: app/src/main/java/com/example/notes/viewmodel/EditScreenViewModel.kt [hunk 3]
```diff
@@ -10,25 +10,25 @@ import com.example.notes.model.NotesManager
 
 class EditScreenViewModel(
     private var navController: NavController,
-    private var notesManager: NotesManager,
+    private var nVM: NavigationViewModel,
```
Comment (jazzyjester) [inline app/src/main/java/com/example/notes/viewmodel/EditScreenViewModel.kt]: @AdiTelem I think that notesManager is better than `nvM`

File: app/src/main/java/com/example/notes/viewmodel/NavigationViewModel.kt [hunk 4]
```diff
@@ -3,14 +3,85 @@ package com.example.notes.viewmodel
...
+import androidx.lifecycle.viewmodel.initializer
+import androidx.lifecycle.viewmodel.viewModelFactory
+import com.example.notes.NotesApplication
+import com.example.notes.R
 import com.example.notes.model.NoteData
+import com.example.notes.model.NoteRepository
 import com.example.notes.model.NotesManager
 
-class NavigationViewModel(context: Context) :
+class NavigationViewModel(var repository: NoteRepository) :
```
Comment (jazzyjester) [inline app/src/main/java/com/example/notes/viewmodel/NavigationViewModel.kt]: `val`

File: app/src/main/java/com/example/notes/model/NoteRepository.kt [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package com.example.notes.model
+
+interface NoteRepository {
+        fun insertNote(noteData: NoteData, callback: (status: Boolean) -> Unit) : Boolean
+        fun readAllNote(callback: (notes: List<NoteData>) -> Unit) : Boolean
+        fun removeNote(noteID: Int, callback: (status: Boolean) -> Unit) : Boolean
+        fun updateNote(noteData: NoteData, callback: (status: Boolean) -> Unit) : Boolean
+}
```
Comment (jazzyjester) [inline app/src/main/java/com/example/notes/model/NoteRepository.kt:8]: @AdiTelem I would use sealed class to encapsulate response, for example:
```
sealed class NetworkResult<T>(
    val data: T? = null,
    val message: String? = null
) {

    class Success<T>(data: T) : NetworkResult<T>(data)

    class Error<T>(message: String?, data: T? = null) : NetworkResult<T>(data, message)

}
```

So you can create your  response that will holds already the data you want to return
Like  : 

`NetworkResult.Success<Boolean>(true)`
`NetworkResult.Failure< NoteData >(message = errorMessage)`

File: app/src/main/java/com/example/notes/model/NotesRemoteDBHelper.kt [hunk 6]
```diff
@@ -6,13 +6,13 @@ import retrofit2.Response
 
 class NotesRemoteDBHelper {
```
Comment (jazzyjester) [inline app/src/main/java/com/example/notes/model/NotesRemoteDBHelper.kt]: is NotesRemoteDBHelper calls the retrofit service ?
if that so, it shouldn't called `NotesRemoteDBHelper`

**Review by jazzyjester — COMMENTED** [review]
Hi @AdiTelem 
I wrote some comments, please take a look...

File: app/src/main/java/com/example/notes/model/NoteRepository.kt (hunk 5 above)
Comment (AdiTelem) [reply app/src/main/java/com/example/notes/model/NoteRepository.kt:8]: note taken, will be back logged and addressed  in further development

**Pull request merged**